### PR TITLE
test: add unit tests for Response Store filter and config

### LINKED ISSUE #230: test: add unit tests for Response Store filter and config
## Context

Coverage is at 90.12%, targeting 95%. The Response Store module has **1,405 lines with zero test coverage**.

## Files needing tests

| File | Lines | What it does |
|------|-------|-------------|
| `apis/src/openai/responses/store/filter.rs` | 914 | Persists non-streaming Responses API responses to SQLite/PostgreSQL with three-phase lifecycle (request, response, body) and handles DELETE operations |
| `apis/src/openai/responses/store/config.rs` | 491 | YAML config validation including database URL security, PostgreSQL SSRF prevention, and SSL certificate validation |

## Suggested test areas

- **filter.rs**: test three-phase persistence lifecycle, DELETE operations, error handling for store failures, streaming vs non-streaming paths
- **config.rs**: test valid/invalid configs, DB URL security validation (SSRF prevention), SSL cert validation, edge cases in PostgreSQL connection string parsing

## Impact

Covering these 1,405 lines would push overall coverage from ~90% toward ~93%.

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Add 47 new tests covering previously untested code paths in the Response Store module (`filter.rs`, `config.rs`, `input_items.rs`)
- Direct `list_input_items` unit tests for pagination edge cases (scalar/null/empty input, limit clamping, cursor logic, ordering)
- Filter lifecycle tests for message assembly edge cases (null/missing output, non-array output, object/null input, custom tenant)
- Config validation tests for SSRF prevention edge cases (IPv6 unique-local/link-local, revalidate_postgres_host, URL fragment parsing, TLS alias path traversal)

Closes #230

## Test plan

- [x] All 192 store module tests pass (`cargo test -p praxis-ai-apis --lib openai::responses::store::tests`)
- [x] Clippy clean (`cargo clippy -p praxis-ai-apis --tests`)

### CODE REVIEW

File: apis/src/openai/responses/store/input_items.rs [hunk 1]
```diff
@@ -132,6 +129,39 @@ pub fn list_input_items(record: &ResponseRecord, params: &ListParams) -> Result<
...
+fn normalize_input_items(record: &ResponseRecord) -> Vec<serde_json::Value> {
+    match &record.input {
+        serde_json::Value::Null => vec![],
+        serde_json::Value::String(text) => vec![scalar_input_item(&record.id, text)],
+        serde_json::Value::Array(arr) => arr.clone(),
+        other => vec![other.clone()],
+    }
+}
+
+/// Build a public input message resource for scalar create input.
```
[the remark this answers is not in the document]
Comment (leseb) [reply apis/src/openai/responses/store/input_items.rs:150]: not worth a comment

File: apis/src/openai/responses/store/input_items.rs [hunk 2]
```diff
@@ -132,6 +129,39 @@ pub fn list_input_items(record: &ResponseRecord, params: &ListParams) -> Result<
...
+/// synthetic user message resource, null input yields an empty list,
+/// and arrays/objects pass through as-is.
+fn normalize_input_items(record: &ResponseRecord) -> Vec<serde_json::Value> {
+    match &record.input {
+        serde_json::Value::Null => vec![],
+        serde_json::Value::String(text) => vec![scalar_input_item(&record.id, text)],
+        serde_json::Value::Array(arr) => arr.clone(),
+        other => vec![other.clone()],
+    }
+}
```
[the remark this answers is not in the document]
Comment (leseb) [reply apis/src/openai/responses/store/input_items.rs:148]: the openai spec has the same asymmetry, you POST a plain string, you GET back a structured resource with id and content: [{"type": "input_text", ...}]

**Review by aslakknutsen — APPROVED** [review]
